### ISSUE TITLE
VSTest task v2  failing Could not load file or assembly 'Microsoft.Teamfoundation.DistributedTask.Task.TestResults

### ISSUE THREAD

**Issue by baywet** [Microsoft/vsts-tasks#4654]
Hi,
I recently upgraded the VSTest task from V1 to v2 in my build definition. When it comes to writing the results, it fails with that output (see bellow).
Agent version: 2.117.1
Custom agent (not hosted pool)
VS2017@latest

```
2017-06-27T13:11:43.6803124Z ##[section]Starting: VsTest - testAssemblies
2017-06-27T13:11:43.6803124Z ==============================================================================
2017-06-27T13:11:43.6803124Z Task         : Visual Studio Test
2017-06-27T13:11:43.6803124Z Description  : Run tests with Visual Studio test runner
2017-06-27T13:11:43.6803124Z Version      : 2.0.58
2017-06-27T13:11:43.6803124Z Author       : Microsoft Corporation
2017-06-27T13:11:43.6803124Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=835764)
2017-06-27T13:11:43.6803124Z ==============================================================================
2017-06-27T13:11:44.2271772Z Run the tests locally using vstest.console.exe
2017-06-27T13:11:44.2271772Z ========================================================
2017-06-27T13:11:44.2271772Z Test selector : Test assemblies
2017-06-27T13:11:44.2271772Z !**\obj\**
2017-06-27T13:11:44.2271772Z Test assemblies : **\*test*.dll,!**\obj\**
2017-06-27T13:11:44.2271772Z Test filter criteria : null
2017-06-27T13:11:44.2271772Z Search folder : C:\agentv2\_work\6\s
2017-06-27T13:11:44.2271772Z Run settings file : C:\agentv2\_work\6\s\MeetingWorkspace.Tests\CodeCoverage.runsettings
2017-06-27T13:11:44.2271772Z Run in parallel : false
2017-06-27T13:11:44.2271772Z Run in isolation : false
2017-06-27T13:11:44.2271772Z Path to custom adapters : null
2017-06-27T13:11:44.2271772Z Other console options : null
2017-06-27T13:11:44.2271772Z Code coverage enabled : true
2017-06-27T13:11:44.2271772Z VisualStudio version selected for test execution : latest
2017-06-27T13:11:44.2271772Z [command]C:\agentv2\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.0.58\vswhere.exe -version [15.0,16.0) -latest -products * -requires Microsoft.VisualStudio.PackageGroup.TestTools.Core -property installationPath
2017-06-27T13:11:44.2428011Z C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise
2017-06-27T13:11:44.2428011Z 
2017-06-27T13:11:44.2428011Z [command]wmic datafile where "name='C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\vstest.console.exe'" get Version /Value
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z Version=15.0.26228.0
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z Version=15.0.26228.0
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z 
2017-06-27T13:11:44.3834512Z ========================================================
2017-06-27T13:11:45.3677872Z [command]C:\agentv2\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.0.58\TestSelector\TestSelector.exe PublishCodeChanges
2017-06-27T13:11:45.4146619Z Is pull request flow: False
2017-06-27T13:11:45.4146619Z 
2017-06-27T13:11:45.4146619Z ***************************** Section 'Get Files that were changed' Starting *****************************
2017-06-27T13:11:46.1991120Z Baseline build: https://2tolead.visualstudio.com/_permalink/_build/index?collectionId=316c93ea-a424-4907-8926-f0f22f909c6f&projectId=188a2d09-bf88-4ba8-a4c9-3e557e709be7&buildId=1102
2017-06-27T13:11:46.1991120Z Baseline build id = 1102
2017-06-27T13:11:46.3090961Z Total files changed = 0
2017-06-27T13:11:46.5741128Z ***************************** Section 'Get Files that were changed' Ended ********************************
2017-06-27T13:11:46.5741128Z 
2017-06-27T13:11:47.3553467Z [command]"C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\MeetingWorkspace.Tests.dll C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions.dll C:\agentv2\_work\6\s\MeetingWorkspace.Tests\bin\Release\Microsoft.VisualStudio.TestPlatform.TestFramework.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.Interface.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\Microsoft.VisualStudio.TestPlatform.TestFramework.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\cs\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\cs\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\cs\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\de\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\de\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\de\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\es\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\es\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\es\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\fr\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\fr\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\fr\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\it\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\it\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\it\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ja\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ja\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ja\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ko\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ko\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ko\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pl\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pl\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pl\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pt\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pt\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\pt\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ru\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ru\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\ru\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\tr\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\tr\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\tr\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hans\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hans\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hans\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hant\Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hant\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\_common\zh-Hant\Microsoft.VisualStudio.TestPlatform.TestFramework.resources.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\netcoreapp1.0\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\uap10.0\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\net45\Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\net45\Microsoft.VisualStudio.TestPlatform.TestFramework.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\netstandard1.0\Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\netstandard1.0\Microsoft.VisualStudio.TestPlatform.TestFramework.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\uap10.0\Microsoft.VisualStudio.TestPlatform.TestFramework.Extensions.dll C:\agentv2\_work\6\s\packages\MSTest.TestFramework.1.1.18\lib\uap10.0\Microsoft.VisualStudio.TestPlatform.TestFramework.dll /ListFullyQualifiedTests /ListTestsTargetPath:C:\agentv2\_work\_temp\2c3103d0-5b3a-11e7-9f06-cfd26cf43bfe.txt "/TestAdapterPath:\"C:\agentv2\_work\6\s\""
2017-06-27T13:11:47.5584670Z Microsoft (R) Test Execution Command Line Tool Version 15.0.26228.0
2017-06-27T13:11:47.5584670Z Copyright (c) Microsoft Corporation.  All rights reserved.
2017-06-27T13:11:47.5584670Z 
2017-06-27T13:11:49.3397206Z Warning: Unable to load types from the test source 'C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\netcoreapp1.0\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll'. Some or all of the tests in this source may not be discovered.
2017-06-27T13:11:49.3397206Z 
2017-06-27T13:11:49.4023159Z Warning: Unable to load types from the test source 'C:\agentv2\_work\6\s\packages\MSTest.TestAdapter.1.1.18\build\uap10.0\Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.dll'. Some or all of the tests in this source may not be discovered.
2017-06-27T13:11:49.4023159Z 
2017-06-27T13:11:49.4250375Z Warning: Input string was not in a correct format.
2017-06-27T13:11:49.4250375Z 
2017-06-27T13:11:49.5187374Z Warning: Input string was not in a correct format.
2017-06-27T13:11:49.5187374Z 
2017-06-27T13:11:49.5343625Z Warning: Input string was not in a correct format.
2017-06-27T13:11:49.5343625Z 
2017-06-27T13:11:49.5656120Z Warning: Input string was not in a correct format.
2017-06-27T13:11:49.5656120Z 
2017-06-27T13:11:49.5812371Z Warning: Input string was not in a correct format.
2017-06-27T13:11:49.5812371Z 
2017-06-27T13:11:50.0656053Z [command]C:\agentv2\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.0.58\TestSelector\TestSelector.exe GetImpactedtests
2017-06-27T13:11:50.2687278Z 
2017-06-27T13:11:50.2687278Z ***************************** Section 'Get Impacted Tests' Starting *****************************
2017-06-27T13:11:51.1750033Z Total impacted tests: 0
2017-06-27T13:11:51.1750033Z No tests were impacted for this build or release.
2017-06-27T13:11:51.1907931Z ***************************** Section 'Get Impacted Tests' Ended ********************************
2017-06-27T13:11:51.1907931Z 
2017-06-27T13:11:51.2062193Z [command]C:\agentv2\_work\_tasks\VSTest_ef087383-ee5e-42c7-9a53-ab56c98420f9\2.0.58\TestSelector\TestSelector.exe UpdateTestResults
2017-06-27T13:11:51.4562141Z 
2017-06-27T13:11:51.4562141Z ***************************** Section 'Update test results' Starting *****************************
2017-06-27T13:11:51.4718396Z 
2017-06-27T13:11:51.4718396Z Exception : Could not load file or assembly 'Microsoft.Teamfoundation.DistributedTask.Task.TestResults, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A)
2017-06-27T13:11:51.4718396Z 
2017-06-27T13:11:51.4718396Z  Stack :    at Microsoft.TestFoundation.TestImpact.TestSelector.CommandUpdateTestResults.Run()
2017-06-27T13:11:51.4718396Z    at Microsoft.TestFoundation.TestImpact.TestSelector.Program.RunCommand(String commandName, String[] args) in E:\v2.0\A5\_work\1\s\Ta\Tasks\TestImpact\TestSelector\Program.cs:line 43
2017-06-27T13:11:51.4718396Z ***************************** Section 'Update test results' Ended ********************************
2017-06-27T13:11:51.4718396Z 
2017-06-27T13:11:51.4718396Z ##[error]Vstest exited with return code: 1.
2017-06-27T13:11:51.4874633Z ##[section]Finishing: VsTest - testAssemblies
```

**Comment by acesiddhu** [comment]
@baywet we are working on this to get fixed. in the mean time you can disable run only impacted test option in the task to get unblocked. Will let you know once this is fixed in production. 
in the mean time can you tell me the vsts account name so that we can track this. you can mail us at devops_tools AT Microsoft DOT com

**Comment by baywet** [comment]
@acesiddhu thanks for the prompt reply, will do.
For now the workaround I found was reverting back to v1.

**Comment by baywet** [comment]
@acesiddhu the email bounced on me, any chance there's a mistake in what you've provided?

**Comment by acesiddhu** [comment]
I got the mail. thanks. will keep u posted

**Comment by acesiddhu** [comment]
@baywet we have deployed a fix for this. let us know if you still face this issue. Sorry for the trouble.

**Comment by baywet** [comment]
@acesiddhu thanks for this fast resolution. I can confirm it's fixed on our end after re-upgrading the task to latest

**Comment by cskwg** [comment]
I had a very similar problem today: the resources DLLs for en-US was missing.
This resources DLL is not part of the package, only the ones for CS, DE, ...TR, ZH,...
Could it be, that the en-US DLLs is missing from the package?
Substituting it with the DE one prevented the error, at least in my tests.
